src/server/scripts/EasternKingdoms/Karazhan/chess_event.cpp

```diff
@@ -0,0 +1,1353 @@
+        {
+            if (eventId == EVENT_CHEAT)
+            {
+                uint32 spellId = 0;
+                switch (urand(0, 2))
+                {
+                case 0:
+                    spellId = (instance->GetData(DATA_TEAM_IN_INSTANCE) == ALLIANCE) ? SPELL_FURY_OF_MEDIVH_ALLIANCE : SPELL_FURY_OF_MEDIVH_HORDE;
+                    break;
+                case 1:
+                    spellId = (instance->GetData(DATA_TEAM_IN_INSTANCE) == ALLIANCE) ? SPELL_HAND_OF_MEDIVH_ALLIANCE : SPELL_HAND_OF_MEDIVH_HORDE;
+                    break;
+                case 2:
+                    spellId = (instance->GetData(DATA_TEAM_IN_INSTANCE) == ALLIANCE) ? SPELL_FULL_HEAL_ALLIANCE : SPELL_FULL_HEAL_HORDE;
+                    break;
+                }
+
+                for (uint8 i = 0; i < urand(1, 3); ++i)
+                    DoCastSelf(spellId, true);
+
+                switch (urand(0, 2))
+                {
+                case 0: DoPlaySoundToSet(me, SOUND_ID_CHEAT_1); break;
+                case 1: DoPlaySoundToSet(me, SOUND_ID_CHEAT_2); break;
+                case 2: DoPlaySoundToSet(me, SOUND_ID_CHEAT_3); break;
+                }
+
+                events.ScheduleEvent(EVENT_CHEAT, 50s, 60s);
+            }
+        }
+    }
+
+    bool OnGossipHello(Player* player) override
+    {
+        InitGossipMenuFor(player, GOSSIP_MENU_ID_MEDIVH);
+        if (instance->GetBossState(DATA_CHESS) == DONE)
+            SendGossipMenuFor(player, GOSSIP_TEXT_ID_MEDIVH_BEATEN, me->GetGUID());
+        else
+            SendGossipMenuFor(player, GOSSIP_TEXT_ID_MEDIVH, me->GetGUID());
+        return true;
+    }
+
+private:
+    EventMap events;
+    InstanceScript* instance;
+};
+
+/*######
+## npc_chess_piece_generic
+######*/
+
+enum Events
+{
+    EVENT_MOVE_COMMAND = 1,
+    EVENT_SPELL_COMMAND,
+    EVENT_MOVE,
+};  
+
+struct npc_chess_piece_generic : public ScriptedAI
+{
+    npc_chess_piece_generic(Creature* creature) : ScriptedAI(creature), instance(creature->GetInstanceScript())
+    {
+        Reset();
+    }
+
+    void Reset() override
+    {
+        me->ApplySpellImmune(0, IMMUNITY_ID, SPELL_GAME_IN_SESSION, true);
+        me->SetSheath(SHEATH_STATE_MELEE);
+        events.Reset();
+        events.ScheduleEvent(EVENT_MOVE_COMMAND, 1s, 2s);
+        if (!me->HasAura(SPELL_CONTROL_PIECE))
+            events.ScheduleEvent(EVENT_SPELL_COMMAND, 1s, 2s);
+        primarySpell = true;
+
+        // cancel move event for player faction npcs or for friendly games
+        if ((instance->GetData(DATA_TEAM_IN_INSTANCE) == ALLIANCE && me->GetFaction() != FACTION_ID_CHESS_HORDE) ||
+            (instance->GetData(DATA_TEAM_IN_INSTANCE) == HORDE && me->GetFaction() != FACTION_ID_CHESS_ALLIANCE) ||
+            instance->GetBossState(DATA_CHESS) == DONE)
+        {
+            events.CancelEvent(EVENT_MOVE_COMMAND);
+        }
+    }
+
+    void AttackStart(Unit* /*victim*/) override { }
```

Comment:
What is the purpose here?

Reply:
When you right click on an enemy chess piece, your piece just starts walking towards it. This is most likely due to the fact that I'm using possess instead of charm. `void AttackStart(Unit* /*victim*/) override { }` prevents this.

Reply:
Yhe method is complete empty and does nothing.

Reply:
In your opinion, then, what does this method do?
https://github.com/TrinityCore/TrinityCore/blob/c70ceb2ee5a4c9952ec5039b477aead75283c4b2/src/server/game/AI/ScriptedAI/ScriptedCreature.cpp#L146-L152